segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -74,12 +80,14 @@
 import org.slf4j.LoggerFactory;
 
 
+import static io.pravega.segmentstore.contracts.Attributes.ATTRIBUTE_SEGMENT_TYPE;
+import static io.pravega.segmentstore.contracts.Attributes.CREATION_TIME;
 import static io.pravega.segmentstore.contracts.Attributes.EVENT_COUNT;
 
 /**
  * Process incoming Append requests and write them to the SegmentStore.
  */
-public class AppendProcessor extends DelegatingRequestProcessor {
+public class AppendProcessor extends DelegatingRequestProcessor implements AutoCloseable {
```

Comment:
Make sure you update **every single unit test** that uses `AppendProcessor` to close it when done. See my comment below in `AppendProcessorTest`